component/loki/source/heroku/heroku.go

```diff
@@ -145,8 +151,15 @@ func (c *Component) Update(args component.Arguments) error {
 			}
 		}
 
+		// [ht.NewHerokuTarget] registers new metrics every time it is called. To
+		// avoid issues with re-registering metrics with the same name, we create a
+		// new registry for the target every time we create one, and pass it to an
+		// unchecked collector to bypass uniqueness checking.
+		registry := prometheus.NewRegistry()
+		c.serverMetrics.SetCollector(serverMetrics)
```

Comment:
```suggestion
		c.serverMetrics.SetCollector(registry)
```
I've missed this in suggestions 😅